frontend/controller/actions/identity-kv.js

```diff
@@ -316,5 +336,56 @@ export default (sbp('sbp/selectors/register', {
 
       await sbp('gi.actions/identity/kv/saveNotificationStatus', { onconflict: getUpdatedNotificationStatus })
     })
+  },
+  // Namespace lookups
+  'gi.actions/identity/kv/fetchCachedNames': async () => {
+    const identityContractID = sbp('state/vuex/state').loggedIn?.identityContractID
+    if (!identityContractID) {
+      throw new Error('Unable to fetch cached names without an active session')
+    }
+    return (await sbp('chelonia/kv/get', identityContractID, KV_KEYS.NS_CACHE))?.data || []
+  },
+  'gi.actions/identity/kv/saveCachedNames': () => {
+    const identityContractID = sbp('state/vuex/state').loggedIn?.identityContractID
+    if (!identityContractID) {
+      throw new Error('Unable to update notification status without an active session')
+    }
+
+    const onconflict = async ({ currentData = [], etag } = {}) => {
+      if (!currentData) currentData = []
+      const data = await checkAndAugmentNames(currentData)
+
+      data.sort()
+      currentData.sort()
+
+      // If there's no difference, there's no point in sending an update
+      if (data.length === currentData.length) {
+        let i = 0
+        for (; i < data.length; i++) {
+          if (data[i] !== currentData[i]) break
+        }
+        if (i !== data.length) return
```

Comment:
I'm not quite understanding the logic here. This is saying that if there's a difference between `data` and `currentData` that we return from this `onconflict` function? If so, that means no update is made because there _is_ a difference? Isn't that backwards?

Or am I misunderstanding? Either way, can you add a comment here explaining?

Reply:
Then shouldn't that be `===` instead of `!==`?